src/common/src/main/java/org/niis/xrd4j/common/util/SOAPHelper.java

```diff
@@ -354,27 +355,32 @@ private static void processMultiMapNode(NodeList list, int index, Map<String, Li
      * have another namespace, the old namespace is first removed and the new
      * namespace is added after that.
      *
-     * @param node Node to be modified
+     * @param node    Node to be modified
      * @param message Message that contains the ProviderMember which namespace
-     * URI and prefix are used
+     * @return changed SOAPElement with added namespace URI and prefix of the ProviderMember
      */
-    public static void addNamespace(Node node, AbstractMessage message) {
+    public static SOAPElement addNamespace(SOAPElement node, AbstractMessage message) {
         if (node.getNodeType() == Node.ELEMENT_NODE) {
-            node.getOwnerDocument().renameNode(node, null, node.getLocalName());
-            if (message.getProducer().getNamespacePrefix() != null
-                    && !message.getProducer().getNamespacePrefix().isEmpty()) {
-                node = (Node) node.getOwnerDocument().renameNode(node, message.getProducer().getNamespaceUrl(),
-                        message.getProducer().getNamespacePrefix() + ":" + node.getNodeName());
-            } else {
-                node = (Node) node.getOwnerDocument().renameNode(node, message.getProducer().getNamespaceUrl(),
-                        node.getNodeName());
+            var soapEl = (SOAPElement) node;
```

Comment:
I think this casting is not needed
```suggestion
            var soapEl = node;
```